### PR TITLE
doc/dev: integrate testing into the narrative

### PR DESCRIPTION
**Pull request by smithfarm**
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

File: doc/dev/index.rst [hunk 1]
```diff
@@ -502,60 +506,99 @@ from 20 minutes to three hours to complete, but it's worth the wait.
...
+A number of integration tests are run on a regular basis in the `Sepia
+lab`_ against the official Ceph repositories (on the master development
+branch and the stable branches). Traditionally, these tests are called "the
+nightlies" because the Ceph core developers used to live and work in
+the same time zone and from their perspective the tests were run overnight. 
 
 The results of the nightlies are visible at either http://pulpito.ceph.com/
 and http://pulpito.ovh.sepia.ceph.com:8081/ (depending on how the tests
-were run) and are also reported on the `ceph-qa mailing list
+were run, i.e. via `teuthology-suite`_ or `teuthology-openstack`_), and are
```
Comment (dachary) [inline doc/dev/index.rst]: I would stick to teuthology-suite for simplicity. I think teuthology-openstack will eventually be hidden from the user by ceph-workbench ceph-qa-suite eventually.

File: doc/dev/index.rst [hunk 2]
```diff
@@ -502,60 +506,99 @@ from 20 minutes to three hours to complete, but it's worth the wait.
...
+<http://docs.ceph.com/teuthology/docs/LAB_SETUP.html>`_ to get you started
+if you decide to go this route.
+
+If you have access to an OpenStack tenant, you have another option:
+teuthology has an OpenStack backend, which is documented `here
+<https://github.com/dachary/teuthology/tree/openstack#openstack-backend>`_.
+While `teuthology-openstack`_ is fairly straightforward, the whole process
+is even easier if one uses `ceph-workbench ceph-qa-suite`_. This tool
+enables the developer to automate deployment of teuthology to an
+OpenStack environment and to control that cloud environment from her local
```
Comment (dachary) [inline doc/dev/index.rst]: her/his ?

File: doc/dev/index.rst [hunk 3]
```diff
@@ -502,60 +506,99 @@ from 20 minutes to three hours to complete, but it's worth the wait.
...
+teuthology-suite
+----------------
 
-Understanding integration tests
-===============================
+Every teuthology cluster, whether bare-metal or cloud-provisioned, has a
+so-called "teuthology node" from which tests suites are triggered using the
+``teuthology-suite`` command. Even if `teuthology-openstack`_ or
+`ceph-workbench ceph-qa-suite`_ is used, ``teuthology-suite`` is run
+behind the scenes.
```
Comment (dachary) [inline doc/dev/index.rst]: I would skip the teuthology-openstack / ceph-workbench part so the reader is focused on teuthology-suite.

File: doc/dev/index.rst [hunk 4]
```diff
@@ -628,25 +709,51 @@ is as follows::
...
+(also known as the "ceph-disk suite") can be run with::
+
+  teuthology-suite --suite ceph-disk
+
+Similarly, an individual test from the ceph-disk suite can be run by
+specifying its description::
 
-To share parts of the test description between suites, the special
-file plus (``+``) can be used to concatenate them. For instance::
+  teuthology-suite --suite ceph-disk/basic/{distros/ubuntu_14.04.yaml tasks/ceph-disk.yaml}
```
Comment (dachary) [inline doc/dev/index.rst]: I don't think that would work. --suite needs to be either a directory or a file. However the following would work

    teuthology-suite --suite ceph-disk --filter 'ceph-disk/basic/{distros/ubuntu_14.04.yaml tasks/ceph-disk.yaml}'

File: doc/dev/index.rst [hunk 5]
```diff
@@ -682,24 +789,16 @@ define the following ``roles``::
...
 
-Each test is uniquely identified by its description which is made of
-the names of all files concatenated together. For instance the test::
+Or a single test from the rbd/thrash suite can be run by
+specifying its description::
 
-  ceph-disk/basic/{distros/centos_7.0.yaml tasks/ceph-disk.yaml}
-
-is the concatenation of the files:
+  teuthology-suite --suite rbd/thrash/{clusters/fixed-2.yaml clusters/openstack.yaml workloads/rbd_api_tests_copy_on_read.yaml}
```
Comment (dachary) [inline doc/dev/index.rst]: s/--suite/--suite rbd --filter=/ for the same reason as above

File: doc/dev/index.rst [hunk 6]
```diff
@@ -744,14 +846,14 @@ reduce the number of tests that are triggered. For instance::
 
 will run as few tests as possible. The tradeoff is that some tests
 will only run on ``ext4`` and not on ``btrfs``, but all files in the
-suite will be in at least one test.
+suite will be in at least one test. FIXME
```
Comment (dachary) [inline doc/dev/index.rst]: s/FIXME/Understanding the actual logic that drives this requires reading the source code./

**Comment by dachary** [comment]
minor things, great improvement over the first version !

File: doc/dev/index.rst (hunk 1 above)
Comment (smithfarm) [inline doc/dev/index.rst]: OK - I dropped teuthology-openstack

File: doc/dev/index.rst (hunk 2 above)
Comment (smithfarm) [inline doc/dev/index.rst]: rephrased

File: doc/dev/index.rst (hunk 3 above)
Comment (smithfarm) [inline doc/dev/index.rst]: Done.

File: doc/dev/index.rst (hunk 4 above)
Comment (smithfarm) [inline doc/dev/index.rst]: Fixed.

File: doc/dev/index.rst (hunk 5 above)
Comment (smithfarm) [inline doc/dev/index.rst]: Fixed.

File: doc/dev/index.rst (hunk 6 above)
Comment (smithfarm) [inline doc/dev/index.rst]: Done.

**Comment by smithfarm** [comment]
@dachary Edited and repushed. Thanks for your review.

**Comment by dachary** [comment]
> documentation-writing mood you could tackle this chapter.

Yes :-)

**Pull request merged**